src/main/java/org/mockito/internal/progress/ThreadSafeMockingProgress.java

```diff
@@ -10,79 +10,112 @@
 import org.mockito.stubbing.OngoingStubbing;
 import org.mockito.verification.VerificationMode;
 import org.mockito.verification.VerificationStrategy;
-
+import java.io.IOException;
+import java.io.ObjectStreamException;
 import java.io.Serializable;
 
-@SuppressWarnings("unchecked")
-public class ThreadSafeMockingProgress implements MockingProgress, Serializable {
-    
-    private static final long serialVersionUID = 6839454041642082618L;
-    private static final ThreadLocal<MockingProgress> mockingProgress = new ThreadLocal<MockingProgress>();
+public class ThreadSafeMockingProgress {
 
-    static MockingProgress threadSafely() {
-        if (mockingProgress.get() == null) {
-            mockingProgress.set(new MockingProgressImpl());
-        }
-        return mockingProgress.get();
-    }
-    
-    public void reportOngoingStubbing(OngoingStubbing<?> iOngoingStubbing) {
-        threadSafely().reportOngoingStubbing(iOngoingStubbing);
-    }
+    private  static MockingProgress SINGLETON = new Impl();
 
-    public OngoingStubbing<?> pullOngoingStubbing() {
-        return threadSafely().pullOngoingStubbing();
-    }
-    
-    public void verificationStarted(VerificationMode verify) {
-        threadSafely().verificationStarted(verify);
+    public final static MockingProgress threadSafeMockingProgress() {
+        return SINGLETON;
```

Comment:
Thats a good idea cause this would avoid  the delegations. Unfortunatly `MockingProgressImpl` is not serializable and I found no other work around for the serialization issues.